core/generate_index.go

```diff
@@ -261,291 +199,66 @@ func (ig *IndexGenerator) DropIndex(bucket []byte) error {
-		}
-	}
-	// By now, the heap has one element for each buffer file
-	batch := ig.db.NewBatch()
-	var nbytes [8]byte
-	for h.Len() > 0 {
-		if err := common.Stopped(ig.quitCh); err != nil {
-			return err
-		}
-
-		element := (heap.Pop(h)).(etl.HeapElem)
-		reader := readers[element.TimeIdx]
-		k := element.Key
-		// Read number of items for this key
-		var count int
-		if n, err := io.ReadFull(reader, nbytes[:]); err != nil || n != 8 {
-			return fmt.Errorf("reading from account buffer file: %d %v", n, err)
-		}
-
-		count = int(binary.BigEndian.Uint64(nbytes[:]))
-		for i := 0; i < count; i++ {
-			var b uint64
-			if n, err := io.ReadFull(reader, nbytes[:]); err != nil || n != 8 {
-				return fmt.Errorf("reading from account buffer file: %d %v", n, err)
-			}
-			b = binary.BigEndian.Uint64(nbytes[:])
-			vzero := (b & emptyValBit) != 0
-			blockNr := b &^ emptyValBit
-
-			currentChunkKey := dbutils.IndexChunkKey(k, ^uint64(0))
-			indexBytes, err1 := batch.Get(bucket, currentChunkKey)
-			if err1 != nil && err1 != ethdb.ErrKeyNotFound {
-				return fmt.Errorf("find chunk failed: %w", err1)
-			}
-
-			var index dbutils.HistoryIndexBytes
-			if len(indexBytes) == 0 {
-				index = dbutils.NewHistoryIndex()
-			} else if dbutils.CheckNewIndexChunk(indexBytes, blockNr) {
-				// Chunk overflow, need to write the "old" current chunk under its key derived from the last element
-				index = dbutils.WrapHistoryIndex(indexBytes)
-				indexKey, err3 := index.Key(k)
-				if err3 != nil {
-					return err3
-				}
-				// Flush the old chunk
-				if err4 := batch.Put(bucket, indexKey, index); err4 != nil {
-					return err4
-				}
-				// Start a new chunk
-				index = dbutils.NewHistoryIndex()
-			} else {
-				index = dbutils.WrapHistoryIndex(indexBytes)
-			}
-			index = index.Append(blockNr, vzero)
-
-			if err := batch.Put(bucket, currentChunkKey, index); err != nil {
-				return err
-			}
-			batchSize := batch.BatchSize()
-			if batchSize > batch.IdealBatchSize() {
-				if _, err := batch.Commit(); err != nil {
-					return err
-				}
-				runtime.ReadMemStats(&m)
-				log.Info("Commited index batch", "bucket", string(bucket), "size", common.StorageSize(batchSize), "current key", fmt.Sprintf("%x...", k[:4]),
-					"alloc", int(m.Alloc/1024), "sys", int(m.Sys/1024), "numGC", int(m.NumGC))
-			}
-		}
-		// Try to read the next key (reuse the element)
-		if n, err := io.ReadFull(reader, element.Key); err == nil && n == keyLength {
-			heap.Push(h, element)
-		} else if err != io.EOF {
-			// If it is EOF, we simply do not return anything into the heap
-			return fmt.Errorf("next reading from account buffer file: %d %x %v", n, element.Key[:n], err)
-		}
-	}
-	if _, err := batch.Commit(); err != nil {
-		return err
+func loadFunc2(k []byte, value []byte, state etl.State, next etl.LoadNextFunc) error { //nolint
+	if len(value)%9 != 0 {
+		log.Error("Strange value", "ln", len(value), "k", common.Bytes2Hex(k))
```

Comment:
hmm, maybe some more info in this log message would be useful for the future debugging